crates/accelerate/src/high_level_synthesis.rs

```diff
@@ -171,82 +213,807 @@ impl QubitTracker {
+    fn __getnewargs__(&self, py: Python) -> PyResult<Py<PyAny>> {
+        (
+            self.hls_config.clone_ref(py),
+            self.hls_plugin_manager.clone_ref(py),
+            self.hls_op_names.clone(),
+            self.coupling_map.clone_ref(py),
+            self.target.clone(),
+            self.equivalence_library.clone(),
+            self.device_insts.clone(),
+            self.use_qubit_indices,
+            self.min_qubits,
+            self.unroll_definitions,
+        )
+            .into_py_any(py)
     }
 
-    /// Extends the correspondence by an additional qubit that
-    /// maps to the given global qubit. Returns the index of the
-    /// new local qubit.
-    fn add_qubit(&mut self, global_qubit: usize) -> usize {
-        let new_local_qubit = self.local_to_global.len();
-        self.local_to_global.push(global_qubit);
-        new_local_qubit
+    fn get_hls_config(&self) -> &Py<PyAny> {
+        &self.hls_config
     }
 
-    /// Returns the local-to-global mapping
-    fn to_global_mapping(&self) -> Vec<usize> {
-        self.local_to_global.clone()
+    fn get_hls_plugin_manager(&self) -> &Py<PyAny> {
+        &self.hls_plugin_manager
     }
 
-    /// Returns the global-to-local mapping
-    fn to_local_mapping(&self) -> HashMap<usize, usize> {
-        HashMap::from_iter(
-            self.local_to_global
-                .iter()
-                .enumerate()
-                .map(|(i, j)| (*j, i)),
+    fn get_hls_op_names(&self) -> Vec<String> {
+        self.hls_op_names.clone()
+    }
+
+    fn get_coupling_map(&self) -> &Py<PyAny> {
+        &self.coupling_map
+    }
+
+    fn get_target(&self, py: Python) -> Option<Py<Target>> {
+        self.target.as_ref().map(|target| target.clone_ref(py))
+    }
+
+    fn get_equivalence_library(&self, py: Python) -> Option<Py<EquivalenceLibrary>> {
+        self.equivalence_library
+            .as_ref()
+            .map(|eqlib| eqlib.clone_ref(py))
+    }
+
+    fn get_device_insts(&self) -> HashSet<String> {
+        self.device_insts.clone()
+    }
+
+    fn get_use_qubit_indices(&self) -> bool {
+        self.use_qubit_indices
+    }
+
+    fn get_min_qubits(&self) -> usize {
+        self.min_qubits
+    }
+
+    fn get_unroll_definitions(&self) -> bool {
+        self.unroll_definitions
+    }
+
+    fn set_device_insts(&mut self, device_insts: HashSet<String>) {
+        self.device_insts = device_insts;
+    }
+
+    fn __str__(&self) -> String {
+        format!(
+            "HighLevelSynthesisData(hls_config: {:?}, hls_plugin_manager: {:?}, hls_op_names: {:?}, coupling_map: {:?}, target: {:?},  equivalence_library: {:?}, device_insts: {:?}, use_qubit_indices: {:?}, min_qubits: {:?}, unroll_definitions: {:?})",
+            self.hls_config, self.hls_plugin_manager, self.hls_op_names, self.coupling_map, self.target, self.equivalence_library, self.device_insts,  self.use_qubit_indices, self.min_qubits, self.unroll_definitions
         )
     }
```

Comment:
You could have this function accept a `Bound` instance of sel, use `getattr` to get the attributes, and use `.repr()?` to get the string representations of the python objects correctly. I also think `Bound` objects do implement `Display` so you you could skip the `repr` call. But that's up to you.